tests/test_table_pixel_size_widget.py

```diff
@@ -0,0 +1,225 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, cast
+
+from qtpy.QtWidgets import QLineEdit
+
+if TYPE_CHECKING:
+    from pytestqt.qtbot import QtBot
+
+
+OBJECTIVE_LABEL = 0
+RESOLUTION_ID = 1
+CAMERA_PX_SIZE = 2
+MAGNIFICATION = 3
+IMAGE_PX_SIZE = 4
+ROW = 0  # "Nikon 40X Plan Fluor ELWD"
+
+
+def _get_wdg(table):
+    obj = table.item(ROW, OBJECTIVE_LABEL).text()
+    resID = cast(QLineEdit, table.cellWidget(ROW, RESOLUTION_ID))
+    mag = cast(QLineEdit, table.cellWidget(ROW, MAGNIFICATION))
+    cam_px = cast(QLineEdit, table.cellWidget(ROW, CAMERA_PX_SIZE))
+    img_px = cast(QLineEdit, table.cellWidget(ROW, IMAGE_PX_SIZE))
+    return obj, resID, mag, cam_px, img_px
+
+
+def test_pixel_size_table(px_wdg):
+    px_size_wdg, table, obj, mmc = px_wdg
+
+    assert ["Res10x", "Res20x", "Res40x"] == list(mmc.getAvailablePixelSizeConfigs())
+    assert table.rowCount() == len(mmc.getStateLabels(obj))
+
+    assert not px_size_wdg.mag_radiobtn.isChecked()
+    assert px_size_wdg.img_px_radiobtn.isChecked()
+
+    obj, resID, mag, cam_px, img_px = _get_wdg(table)
+
+    assert obj == "Nikon 40X Plan Fluor ELWD"
+    assert resID.text() == "Res40x"
+    assert resID.property("resID") == "Res40x"
+    assert resID.graphicsEffect().opacity() == 1.00
+    assert mag.text() == "40.0"
+    assert mag.graphicsEffect().opacity() == 1.00
+    assert cam_px.text() == "10.00"
+    assert cam_px.graphicsEffect().opacity() == 1.00
+    assert img_px.text() == "0.2500"
+    assert img_px.graphicsEffect().opacity() == 1.00
+    assert img_px.styleSheet() == "color:magenta"
+
+    for r in range(table.rowCount()):
+        _cam_px = table.cellWidget(r, CAMERA_PX_SIZE).text()
+        assert _cam_px == "10.00"
+
+
+def test_change_magnification(qtbot: QtBot, px_wdg):
+    _, table, _, mmc = px_wdg
+
+    _, _, mag, cam_px, img_px = _get_wdg(table)
+
+    mag.setText("50.0")
+    with qtbot.waitSignal(mag.editingFinished):
+        mag.editingFinished.emit()
```

Comment:
these patterns (where you await the very signal you are explicitly emitting) are a bit strange.  did you come to this out of necessity?

Normally, `waitSignal` is used when you expect some signal to mark the end of some other action:

```python
with qtbot.waitSignal(widget.valueChanged):
    widget.push_unrelated_button()
```

so, why isn't it sufficient just to use `mag.setText("50.0")`?   if someone programatically did that and then didn't follow it up with `editingFinished.emit()` would things be out of sync?

Reply:
@tlambert03  I've use this pattern because `wdg.setText()` does not emit the `editingFinished` signal. I preferred to use the use the `editingFinished` signal for the QLineEdit table widgets instead of the `textChanged` signal so that the signal is emitted only when the user press enter and not every time is typing. 
Do you think we should I switch to the `textChanged` signal?

Reply:
nah, the goal of not over-emitting on every letter is definitely a good one.  I suppose we can just leave the potential "buggish" thing of someone manipulating the GUI programatically with `setText`... or at least cross that bridge later